cri/stream/runtime.go

```diff
@@ -0,0 +1,164 @@
+package stream
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	apitypes "github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/cri/stream/remotecommand"
+	"github.com/alibaba/pouch/daemon/mgr"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Runtime is the interface to execute the commands and provide the streams.
+type Runtime interface {
+	// Exec executes the command in pod.
+	Exec(containerID string, cmd []string, streamOpts *remotecommand.Options, streams *remotecommand.Streams) (uint32, error)
+
+	// Attach attaches to pod.
+	Attach(containerID string, streamOpts *remotecommand.Options, streams *remotecommand.Streams) error
+
+	// PortForward forward port to pod.
+	PortForward(name string, port int32, stream io.ReadWriteCloser) error
+}
+
+type streamRuntime struct {
+	containerMgr mgr.ContainerMgr
+}
+
+// NewStreamRuntime creates a brand new stream runtime.
+func NewStreamRuntime(ctrMgr mgr.ContainerMgr) Runtime {
+	return &streamRuntime{containerMgr: ctrMgr}
+}
+
+// Exec executes a command inside the container.
+func (s *streamRuntime) Exec(containerID string, cmd []string, streamOpts *remotecommand.Options, streams *remotecommand.Streams) (uint32, error) {
```

Comment:
can we add the `context` as first argument? If we have the `ctx`, we can handle the progress by the upstream.